arch/rv32i/src/pmp.rs

```diff
@@ -478,148 +490,161 @@ impl kernel::mpu::MPU for PMPConfig {
-                            csr::CSR.pmpaddr1.set((start as u32 + size as u32) >> 2);
-                        }
-                        1 => {
-                            // Disable access up to the start address
-                            csr::CSR.pmpcfg0.modify(
-                                csr::pmpconfig::pmpcfg::r2::CLEAR
-                                    + csr::pmpconfig::pmpcfg::w2::CLEAR
-                                    + csr::pmpconfig::pmpcfg::x2::CLEAR
-                                    + csr::pmpconfig::pmpcfg::a2::TOR,
-                            );
-                            csr::CSR.pmpaddr2.set((start as u32) >> 2);
-
-                            // Set access to end address
-                            csr::CSR.pmpcfg0.set(cfg_val << 24 | csr::CSR.pmpcfg0.get());
-                            csr::CSR.pmpaddr3.set((start as u32 + size as u32) >> 2);
-                        }
-                        2 => {
-                            // Disable access up to the start address
-                            csr::CSR.pmpcfg1.modify(
-                                csr::pmpconfig::pmpcfg::r0::CLEAR
-                                    + csr::pmpconfig::pmpcfg::w0::CLEAR
-                                    + csr::pmpconfig::pmpcfg::x0::CLEAR
-                                    + csr::pmpconfig::pmpcfg::a0::TOR,
-                            );
-                            csr::CSR.pmpaddr4.set((start as u32) >> 2);
-
-                            // Set access to end address
-                            csr::CSR.pmpcfg1.set(cfg_val << 8 | csr::CSR.pmpcfg0.get());
-                            csr::CSR.pmpaddr5.set((start as u32 + size as u32) >> 2);
-                        }
-                        3 => {
-                            // Disable access up to the start address
-                            csr::CSR.pmpcfg1.modify(
-                                csr::pmpconfig::pmpcfg::r3::CLEAR
-                                    + csr::pmpconfig::pmpcfg::w3::CLEAR
-                                    + csr::pmpconfig::pmpcfg::x3::CLEAR
-                                    + csr::pmpconfig::pmpcfg::a3::TOR,
-                            );
-                            csr::CSR.pmpaddr6.set((start as u32) >> 2);
-
-                            // Set access to end address
-                            csr::CSR.pmpcfg1.set(cfg_val << 24 | csr::CSR.pmpcfg0.get());
-                            csr::CSR.pmpaddr7.set((start as u32 + size as u32) >> 2);
+    fn configure_mpu(&self, config: &Self::MpuConfig, app_id: &AppId) {
+        let last_configured_for_this_app = self
+            .last_configured_for
+            .map_or(false, |last_app_id| last_app_id == app_id);
+
+        // Skip PMP configuration if it is already configured for this app and the MPU
+        // configuration of this app has not changed.
+        if !last_configured_for_this_app || config.is_dirty.get() {
+            // tock/tock#1860
```

Comment:
Can you remove this?

Reply:
Hmm, I thought the comment and the PR description were pretty clear. This skips PMP configuration if it is already configured for the given app and has not changed since the last time it was committed. Please see also #1730 and #1860.

Reply:
The comment does that?

Reply:
Just to be clear, are you referring to just `tock/tock#1860` or the `if` statement? It is hard to tell looking at the way your comment is rendered on github (my reply above is for the latter but probably you meant the former). I added a reference to #1860 because the regions are sorted every time PMP is configured.

Reply:
Yeah, I was referring to the comment. The GitHub UI is really lacking unfortunately, so it can be confusing.

Can we remove the comment then? I don't think it adds anything and we have a tracking issue to track the inefficiently.

Reply:
Ah, apologies for the confusion (I know github shows some context above the line that you comment but I was confused because this one is perfectly aligned). Done.

Reply:
Thanks! It's good to have someone else looking at the PMP code :)

Reply:
I think it might be useful to refer to the Github issue in the PR description, and instead add a descriptive comment here.